controllers/cloud.redhat.com/handlers.go

```diff
@@ -111,7 +155,58 @@ func (e *enqueueRequestForObjectCustom) Create(evt event.CreateEvent, q workqueu
 	}
 }
 
+func (e *enqueueRequestForObjectCustom) doUpdateToHash(obj client.Object, q workqueue.RateLimitingInterface) error {
+	e.logMessage(obj, "update needed because changed", "", getNamespacedName(obj))
+	hashObj, err := e.hashCache.Read(obj)
+	if err != nil {
+		e.logMessage(obj, err.Error(), "", getNamespacedName(obj))
+		return err
+	}
+
+	var loopObjs map[types.NamespacedName]bool
+	switch e.TypeOfOwner.(type) {
+	case *crd.ClowdApp:
+		loopObjs = hashObj.ClowdApps
+	case *crd.ClowdEnvironment:
+		loopObjs = hashObj.ClowdEnvs
+	}
+
+	for k := range loopObjs {
+		q.Add(reconcile.Request{NamespacedName: k})
+	}
+	return nil
+}
+
+func (e *enqueueRequestForObjectCustom) reconcileAllAppsUsingObject(obj client.Object, q workqueue.RateLimitingInterface) {
+	capps := &crd.ClowdAppList{}
+	if err := e.client.List(e.context, capps, client.InNamespace(obj.GetNamespace())); err != nil {
+		e.logMessage(obj, err.Error(), "error listing apps", getNamespacedName(obj))
+	}
+	for _, app := range capps.Items {
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{Name: app.GetName(), Namespace: obj.GetNamespace()}})
+	}
+}
+
 func (e *enqueueRequestForObjectCustom) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	if evt.ObjectNew.GetAnnotations()["qontract.reconcile"] == "true" {
+		shouldUpdate, err := e.updateHashCacheForConfigMapAndSecret(evt.ObjectNew)
+		e.logMessage(evt.ObjectNew, "debug", fmt.Sprintf("shouldUpdate %s %v", e.ctrlName, shouldUpdate), getNamespacedName(evt.ObjectNew))
+		if err != nil {
+			e.logMessage(evt.ObjectNew, err.Error(), "", getNamespacedName(evt.ObjectNew))
+		}
+		if shouldUpdate {
+			_ = e.doUpdateToHash(evt.ObjectNew, q)
+		}
+		if evt.ObjectOld.GetAnnotations()["qontract.reconcile"] != evt.ObjectNew.GetAnnotations()["qontract.reconcile"] {
```

Comment:
Same comment as above - I'm surprised to see a specific annotation baked into the implementation. Why is this carved out in so many places? What makes this annotation special?

Reply:
see above